### DISCUSSION TITLE
How can I check if a test has been created for a specific class?

### DISCUSSION THREAD

**Discussion by CharlesMoreira1** [LemonAppDev/konsist#627] [Ask a Question] [answered]
### Wanted end result.

Hello!

I recently discovered your library, and I found it amazing. I haven't tested it yet, but a question came to mind. Would it be possible to check if a test has been created for a specific view?

For example, I have an app that performs snapshot tests, and I would like to ensure that snapshot tests have been created for that view, either by package or by the "SnapshotTest.kt" suffix.

For example: "NameViewSnapshotTest".

Is this feasible?

### What you tried so far.

I attempted to use the Danger System to establish a rule in Ruby. This way, my GitHub actions would signal, and a bot would initiate a thread in the pull request (PR).

**Comment by igorwojda** [comment, marked as answer]
We have a `hasTest` simple method that checks is test exists for a given class:

```kotlin
@Test
fun `every class has test`() {
    Konsist
        .scopeFromProduction()
        .classes()
        .assert { it.hasTestClass(testFileNameSuffix = "SnapshotTest") }
}
```

**Reply by CharlesMoreira1** [reply]
That's perfect! 

It's exactly what I was looking for. 

However, when adding this test to a project that's already in progress, I would need to add a suppress for the Views that don't yet have a snapshot test, right?

**Reply by igorwojda** [reply]
There are two ways of doing this
1. Suppress https://docs.konsist.lemonappdev.com/writing-tests/suppressing-konsist-test
2. Define a more granular scope https://docs.konsist.lemonappdev.com/writing-tests/koscope

**Reply by CharlesMoreira1** [reply]
It's perfect! Thanks!
